currency_rate_update/models/res_currency_rate.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
+
+
+class ResCurrencyRate(models.Model):
+    _name = 'res.currency.rate'
+    _inherit = ['res.currency.rate', 'mail.thread']
+
+    rate = fields.Float(
+        track_visibility='onchange',
+    )
+    provider_id = fields.Many2one(
+        string='Provider',
+        comodel_name='res.currency.rate.provider',
+        ondelete='restrict',
+        track_visibility='onchange',
+    )
+
+    @api.multi
+    def write(self, values):
+        if 'rate' in values and 'provider_id' not in values:
+            values['provider_id'] = False
```

Comment:
Why do you need this?

Reply:
If set manually - remove the reference to provider. Since user is the provider

Reply:
OK, please put this as docstring in the method (and it's interesting to always do it on all overwrites that can require an explanation)

Reply:
What happen if you change the date?

Reply:
Valid point